presto-parser/src/main/java/com/facebook/presto/sql/parser/ParsingOptions.java

```diff
@@ -25,19 +29,65 @@
     }
 
     private final DecimalLiteralTreatment decimalLiteralTreatment;
+    private final Consumer<ParsingWarning> warningConsumer;
 
+    // This constructor is deprecated. Use the Builder instead.
+    @Deprecated
     public ParsingOptions()
     {
         this(DecimalLiteralTreatment.REJECT);
     }
 
+    // This constructor is deprecated. Use the Builder instead.
+    @Deprecated
     public ParsingOptions(DecimalLiteralTreatment decimalLiteralTreatment)
+    {
+        this(decimalLiteralTreatment, NOOP_WARNING_CONSUMER);
+    }
+
+    private ParsingOptions(DecimalLiteralTreatment decimalLiteralTreatment, Consumer<ParsingWarning> warningConsumer)
     {
         this.decimalLiteralTreatment = requireNonNull(decimalLiteralTreatment, "decimalLiteralTreatment is null");
+        this.warningConsumer = requireNonNull(warningConsumer, "warningConsumer is null");
     }
 
     public DecimalLiteralTreatment getDecimalLiteralTreatment()
     {
         return decimalLiteralTreatment;
     }
+
+    public Consumer<ParsingWarning> getWarningConsumer()
+    {
+        return warningConsumer;
+    }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
```

Comment:
Why do you need to make this `public` if you have the static `builder()` method?

Reply:
The Builder methods need to be accessible: ex. QueryPreparer and LocalQueryRunner.